addons/website/models/mixins.py

```diff
@@ -633,6 +633,36 @@ def _search_build_domain(self, domain_list, search, fields, extra=None):
                 domain &= Domain.OR(subdomains)
         return domain
 
+    @api.model
+    def _search_build_custom_query(self, domain_list, search, fields, model, extra=None):
+        domain = Domain.AND(domain_list)
+        if search:
+            table = model._table
+            search_terms = [f"'%{t}%'" for t in search.split()]
+            required_number_of_search_terms = len(search_terms)
+            if len(search_terms) <= 5:
+                required_number_of_search_terms = len(search_terms) - 1
+            else:
+                required_number_of_search_terms = len(search_terms) - 2
+            fields_with_table_name = [f"{table}.{field}" for field in fields]
+            outer = ""
+            for search_term in search_terms:
+                inner = ""
+                for field in fields_with_table_name:
+                    inner += f"({field}::text ILIKE {search_term})::int,"
+                inner = inner[:-1]  # remove last ','
+                outer += f"GREATEST({inner}) + "
+            outer = outer[:-3]  # remove last ' + '
+            query = f"""
+                SELECT id FROM {table}
+                WHERE ({outer}) >= {required_number_of_search_terms}
```

Comment:
Query will generate x number of GREATEST check where x will be number of search terms, can you improve query(I have shared some suggestions given in personal discord)